comm/p2p/libp2p.go

```diff
@@ -105,42 +114,70 @@ func (c Libp2pCommunication) UnSubscribe(
 
 /** Helper methods **/
 
+func (c Libp2pCommunication) StreamHandlerFunc(s network.Stream) {
+	c.ProcessMessagesFromStream(s)
```

Comment:
If some panic happens here stream will remain unclosed and there is no way to recover